python/ray/includes/unique_ids.pxi

```diff
@@ -0,0 +1,903 @@
+"""This is a module for unique IDs in Ray.
+We define different types for different IDs for type safe.
+
+See https://github.com/ray-project/ray/issues/3721.
```

Comment:
One reason I am defining those IDs as independent classes is because those C++ types could be strong-typedefed in the near future.
I have also tried to reuse some duplicated parts. Unfortunately, Cython cannot recognize classes from `cls` or `__class__` in many cases. So I still need to write a lot of extra code.

Reply:
Do you see any other way to get rid of the code duplication? It would be good to resolve this before the PR is merged.

Reply:
I agree that all of these types will be different C++ classes in the future, so we should plan for that setting. It's also possible that they will have different methods, e.g, with `ObjectID` you could potentially call `GetTaskID()` or `IsPut()`, but that wouldn't make sense with an `ActorID`.

Reply:
Agreed, but this doesn't mean they can't share code. I'm pretty sure with interitence it can be made to work well that the common methods are defined only once, I'll give it a shot.